### PR TITLE
Asman Umbetov Tasks 41-50

### PR DESCRIPTION
**Pull request by fastndead**

### CODE REVIEW

File: tasks/41_sort-by-bits/sort-by-bits.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export function sortByBits(arr) {
+  return [...arr].sort((a, b) => {
+    return bitCount(a) - bitCount(b) === 0 ? a - b : bitCount(a) - bitCount(b);
+  } );
```
Comment (vovcyan) [inline tasks/41_sort-by-bits/sort-by-bits.js]: Remove space

File: tasks/45_equal-sides/equal-sides.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export function findEqualIndex(arr) {
+  for (let i = 0; i < arr.length; i++) {
+    if (sum(arr.slice(0, i)) === sum(arr.slice(i + 1, arr.length))) {
```
Comment (vovcyan) [inline tasks/45_equal-sides/equal-sides.js]: Split it

File: tasks/48_follow-the-spy/follow-the-spy.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export function followTheSpy(arr) {
+  const route = [arr[0][0]];
```
Comment (vovcyan) [inline tasks/48_follow-the-spy/follow-the-spy.js]: Split it

File: tasks/48_follow-the-spy/follow-the-spy.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  for (let i = 0; i < arr.length; i++) {
+    route.push(arr.find((item) => {
+      return item[0] === route[route.length - 1];
+    })[1]);
```
Comment (vovcyan) [inline tasks/48_follow-the-spy/follow-the-spy.js]: Rewrite this logic please

File: tasks/49_cat-and-dog-years/cat-and-dog-years.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export function humanYearsCatYearsDogYears(humanYears) {
+  if (humanYears === 1) {
+    return [1, 15, 15];
+  }
+  if (humanYears === 2) {
+    return [2, 24, 24];
+  }
+  return [humanYears,
```
Comment (vovcyan) [inline tasks/49_cat-and-dog-years/cat-and-dog-years.js]: Move it to new line

File: tasks/49_cat-and-dog-years/cat-and-dog-years.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    24 + (humanYears - 2) * 4,
+    24 + (humanYears - 2) * 5];
```
Comment (vovcyan) [inline tasks/49_cat-and-dog-years/cat-and-dog-years.js]: Move ]; to new line

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/48_follow-the-spy/follow-the-spy.js (hunk 3 above)
Comment (fastndead) [reply tasks/48_follow-the-spy/follow-the-spy.js]: what do you mean split it? should I make it like this?: 
Снимок экрана 2019-09-16 в 12 11 34

File: tasks/48_follow-the-spy/follow-the-spy.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export function followTheSpy(arr) {
+  const initialPoint = arr[0][0];
+  const route = [initialPoint];
+
+  for (let i = 0; i < arr.length; i++) {
+    let nextItinerary = arr.find((item) => {
+      return item[0] === route[route.length - 1];
+    });
+    let [, nextRoutePoint] = nextItinerary;
```
Comment (vovcyan) [inline tasks/48_follow-the-spy/follow-the-spy.js]: const

**Review by vovcyan — CHANGES_REQUESTED** [review]
Remove this file from your PR tasks/56_search-and-replace/search-and-replace.js

**Review by vovcyan — DISMISSED** [review]

File: tasks/42_first-n-smallest/first-n-smallest.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export function firstNSmallest(arr, num) {
+  const sortedArr = [...arr].sort();
+  return arr.filter((item) => {
+    return sortedArr.slice(0, num).includes(item);
```
Comment (alexkond) [inline tasks/42_first-n-smallest/first-n-smallest.js]: You're creating a new array on each iteration. This is not optimal. Consider creating this array out of iteration

File: tasks/44_time-simulation/time-simulation.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+export function SimTime() {
+  let simTime = 0;
+  let realTime = 0;
+  let speed = 1;
+
+  this.get = function() {
```
Comment (alexkond) [inline tasks/44_time-simulation/time-simulation.js]: Adding methods inside the constructor function has some drawbacks. Please use the prototype.

**Review by alexkond — CHANGES_REQUESTED** [review]

**Review by vovcyan — DISMISSED** [review]

File: tasks/44_time-simulation/time-simulation.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+export function SimTime() {
+  let simTime = 0;
+  let realTime = 0;
+  let speed = 1;
+
+  SimTime.prototype.get = function() {
+    return simTime;
+  };
+
+  SimTime.prototype.setSpeed = function(newSpeed) {
```
Comment (savchukm) [inline tasks/44_time-simulation/time-simulation.js]: `this. setSpeed = ...` will be enough

File: tasks/47_string-expansion/string-expansion.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+export function stringExpansion(str) {
```
Comment (savchukm) [inline tasks/47_string-expansion/string-expansion.js]: Usually export function goes as a last one defined in a file

**Review by savchukm — CHANGES_REQUESTED** [review]

File: tasks/44_time-simulation/time-simulation.js (hunk 10 above)
Comment (fastndead) [reply tasks/44_time-simulation/time-simulation.js]: I was requested to use prototypes by alexkond, though

Comment (alexkond) [reply tasks/44_time-simulation/time-simulation.js]: Yes, but it should be done outside of the constructor function.

Comment (savchukm) [reply tasks/44_time-simulation/time-simulation.js]: Ok

**Review by vovcyan — APPROVED** [review]

**Review by alexkond — APPROVED** [review]

**Pull request merged**